conformance/lang/expressions/checking-expression/check_expr.balt

```diff
@@ -0,0 +1,2020 @@
+
+    error? x2 = fn2(); // @output 12.3
+                       // @output 9.223372036854776E18
+                       // @output -92233.82036854776
+                       // @output 18.670913696289062
+                       // @output -171.75440979003906
+
+    error? x3 = fn3(); // @output 12.3
+                       // @output 9223372036854775807
+                       // @output -92233.82036854775807
+
+    error? x4 = fn4(); // @output
+                       // @output 9223372036854775807d
+
+    error? x5 = fn5(); // @output
+                       // @output ()
+}
+
+function fn1() {
+    int a1 = check 123;
+    io:println(a1);
+    int a2 = check 9223372036854775807;
+    io:println(a2);
+    int a3 = check (-9223372036854775807 - 1);
+    io:println(a3);
+    int a4 = check 0x12abC1;
+    io:println(a4);
+    int a5 = check -0xAbc121;
+    io:println(a5);
+}
+
+function fn2() {
+    float a1 = check 12.3;
+    io:println(a1);
+    float a2 = check 9223372036854775807;
+    io:println(a2);
+    float a3 = check (-92233.72036854775807 - .1);
+    io:println(a3);
+    float a4 = check 0x12.abC1;
+    io:println(a4);
+    float a5 = check -0xAb.c121;
+    io:println(a5);
+}
+
+function fn3() {
+    decimal a1 = check 12.3d;
+    io:println(a1);
+    decimal a2 = check 9223372036854775807d;
+    io:println(a2);
+    decimal a3 = check (-92233.72036854775807d - .1d);
+    io:println(a3);
+}
+
+function fn4() {
+    string a1 = check "";
+    io:println(a1);
+    string a2 = check "9223372036854775807d";
+    io:println(a2);
+}
+
+function fn5() {
+    () a1 = check ();
+    io:println(a1);
+    () a2 = check null;
+    io:println(a2.toBalString());
+}
+
+Test-Case: output
+Description: Test check-expr with string template expressions.
+Labels: check, string, function-defn, function-call-expr, return-stmt, optional-type, DecimalNumber, string-template-expr,
+        int, trap
+
+function init() {
+    error? x1 = fn1(); // @output 224
+                       // @output ab
+                       // @output a
+
+    error|string x2 = fn2(2, 4);
+    io:println(x2); // @output 1
+
+    error|string x3 = fn2(2, 0);
+    io:println(x3); // @output error("{ballerina}DivisionByZero",message=" / by zero")
+}
+
+function fn1() returns error? {
+    string a1 = check (string`${12 + 212 - 0}`);
+    io:println(a1);
+    string a2 = check (string`${"a" + "b"}`);
+    io:println(a2);
+    string a3 = check (string`${string `${"a"}`}`);
+    io:println(a3);
+}
+
+function fn2(int a, int b) returns error|string {
+    return  string`${check (trap a / b) + 1}`;
+}
+
+Test-Case: error
+Description: Test the static type of the check-expr when it is used with string template expressions.
+Labels: decimal, DecimalNumber, float, string, function-defn, function-call-expr, return-stmt, optional-type, check,
+        string-template-expr, error-type, int
+
+function errorFunction() {
+    error? x1 = fn1(); // @error expected '(int|error)' but found 'string'
```

Comment:
This error will be thrown inside the `fn1()` function and won't be given as a output of the function.

Reply:
My bad. Will fix.